zone/npc.cpp

```diff
@@ -1469,324 +1474,351 @@ NPC* NPC::SpawnNPC(const char* spawncommand, const glm::vec4& position, Client*
+	e.gender       = n->GetGender();
+	e.texture      = n->GetTexture();
+	e.helmtexture  = n->GetHelmTexture();
+	e.size         = n->GetSize();
+	e.loottable_id = n->GetLoottableID();
+	e.merchant_id  = n->MerchantType;
+	e.face         = n->GetLuclinFace();
 
-	std::string query = StringFormat("SELECT id, spawngroupID FROM spawn2 WHERE "
-					 "zone='%s' AND spawngroupID=%i",
-					 zone, spawn->GetSpawnGroupId());
-	auto results = QueryDatabase(query);
-	if (!results.Success())
-		return 0;
+	const int updated = NpcTypesRepository::UpdateOne(*this, e);
 
-	if (results.RowCount() == 0)
-		return 0;
+	return updated;
+}
 
-	auto row = results.begin();
-	if (row[0])
-		id = Strings::ToInt(row[0]);
+uint32 ZoneDatabase::DeleteSpawnLeaveInNPCTypeTable(const std::string& zone, Client* c, NPC* n)
+{
+	const auto& l = Spawn2Repository::GetWhere(
+		*this,
+		fmt::format(
+			"`zone` = '{}' AND `spawngroupID` = {}",
+			zone,
+			n->GetSpawnGroupId()
+		)
+	);
 
-	if (row[1])
-		spawngroupID = Strings::ToInt(row[1]);
+	if (l.empty()) {
+		return 0;
+	}
 
-	query = StringFormat("DELETE FROM spawn2 WHERE id = '%i'", id);
-	results = QueryDatabase(query);
-	if (!results.Success())
+	if (!Spawn2Repository::DeleteOne(*this, l[0].id)) {
 		return 0;
+	}
 
-	query = StringFormat("DELETE FROM spawngroup WHERE id = '%i'", spawngroupID);
-	results = QueryDatabase(query);
-	if (!results.Success())
+	if (!SpawngroupRepository::DeleteOne(*this, l[0].spawngroupID)) {
 		return 0;
+	}
 
-	query = StringFormat("DELETE FROM spawnentry WHERE spawngroupID = '%i'", spawngroupID);
-	results = QueryDatabase(query);
-	if (!results.Success())
+	if (!SpawnentryRepository::DeleteOne(*this, l[0].spawngroupID)) {
 		return 0;
+	}
 
 	return 1;
 }
 
-uint32 ZoneDatabase::DeleteSpawnRemoveFromNPCTypeTable(const char *zone, uint32 zone_version, Client *client,
-						       NPC *spawn)
+uint32 ZoneDatabase::DeleteSpawnRemoveFromNPCTypeTable(
+	const std::string& zone,
+	uint32 instance_version,
+	Client* c,
+	NPC* n
+)
 {
-	uint32 id = 0;
-	uint32 spawngroupID = 0;
-
-	std::string query = StringFormat("SELECT id, spawngroupID FROM spawn2 WHERE zone = '%s' "
-					 "AND (version = %u OR version = -1) AND spawngroupID = %i",
-					 zone, zone_version, spawn->GetSpawnGroupId());
-	auto results = QueryDatabase(query);
-	if (!results.Success())
-		return 0;
+	const auto& l = Spawn2Repository::GetWhere(
+		*this,
+		fmt::format(
+			"`zone` = '{}' AND (`version` = {} OR `version` = -1) AND `spawngroupID` = {}",
+			zone,
+			instance_version,
+			n->GetSpawnGroupId()
+		)
+	);
 
-	if (results.RowCount() == 0)
+	if (l.empty()) {
 		return 0;
+	}
 
-	auto row = results.begin();
-
-	if (row[0])
-		id = Strings::ToInt(row[0]);
-
-	if (row[1])
-		spawngroupID = Strings::ToInt(row[1]);
-
-	query = StringFormat("DELETE FROM spawn2 WHERE id = '%i'", id);
-	results = QueryDatabase(query);
-	if (!results.Success())
+	if (!Spawn2Repository::DeleteOne(*this, l[0].id)) {
```

Comment:
Let's use a different receiver other than `l[0]` for readability

```cpp
auto s = l[0];
```

Use s below